components/core/src/fs.rs

```diff
@@ -822,8 +821,8 @@ mod test_find_command {
     fn setup_path() {
```

Comment:
`locked_env_var` is defined in `common` so we would have to move it to core and adjust all callers which feels like it belongs in a separate PR.

Reply:
Yeah, it's overdue to be moved into `core` (probably common/core should be merged entirely), but I think it's worth doing now since these tests are prone to non-deterministic failures without it. A separate PR is good, but it could merge before this one so we can base the fix on it.

How about taking the dependency-injection approach? That's the preferable solution in my view and is appropriate to this PR scope.

Reply:
Including that in here feels to me kinda like scope creep for this PR but I'll defer to you both on it!

Reply:
I agree that moving `locked_env_var` deserves its own PR if we go that way. If we use dependency injection instead, we can make those changes in this PR and leave moving `locked_env_var` for later, but either way I think we should not leave this in a subtly broken state which may crop up some unknown time in the future.

Reply:
In this case, I think even if multiple tests concurrently change this, its harmless. worse that will happen id the fixtures will get added to the path multiple times.

Reply:
I'm not so sure we can make that strong an assumption about the way `env::set_var` works. The documentation says "concurrent access to environment variables is safe in Rust", but that doesn't mean that it will do what you expect. I'm not sure we're guaranteed one thread accessing this in the midst of another thread setting it couldn't result in an empty result.

In any case, it's not a huge deal, but it's an error-prone pattern to rely on global state like this, so when we can switch to something more robust (such as injecting global dependencies as arguments) without a huge hassle, I think we'd be well served to do so.

Reply:
Looking at this I still think it is out of scope. These tests while not ideal has been in place for 3 years without raising stability issues. I'd just not go into refactoring non-test code unrelated to the intent of this PR and get it merged.

Reply:
In test code, I'm equally or more worried about false positives, but you're right that we can probably let this one go. The value of getting this merged is quite high.